src/test/java/RandomGeneratorTest.java

```diff
@@ -0,0 +1,42 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+class RandomGeneratorTest {
+    RandomGenerator randomGenerator;
+    TestClass testClass;
+
+    @BeforeEach
+    void setUp() {
+        randomGenerator=new RandomGenerator();
+        testClass = new TestClass();
+    }
+
+    @Test
+    void generateRandomObjectForType() throws  InvocationTargetException, IllegalAccessException, NoSuchFieldException {
```

Comment:
generateRandomObjectForType will no longer be public "ig", so should test only generateRandomObject